File: src/test/java/doc/java/examples/filters/MyTextCommandFilter.java
package doc.java.examples.filters;

import io.github.freya022.botcommands.api.commands.text.TextCommandFilter;
import io.github.freya022.botcommands.api.commands.text.TextCommandVariation;
import io.github.freya022.botcommands.api.core.service.annotations.BService;
import io.github.freya022.botcommands.test.switches.TestLanguage;
import io.github.freya022.botcommands.test.switches.TestService;
import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@BService
@TestService
@TestLanguage(TestLanguage.Language.JAVA)
public class MyTextCommandFilter implements TextCommandFilter {

    @Override
    public boolean getGlobal() {
        return true;
    }

    @Nullable
    @Override
    public String check(@NotNull MessageReceivedEvent event, @NotNull TextCommandVariation commandVariation, @NotNull String args) {
        if (event.getChannel().getIdLong() != 722891685755093076L) {
            event.getMessage().reply("Can only run commands in <#722891685755093076>").queue();
            return "Wrong channel";
        }
        return null;
    }
}

File: src/test/kotlin/doc/kotlin/examples/filters/MyTextCommandFilter.kt
package doc.kotlin.examples.filters

import dev.minn.jda.ktx.coroutines.await
import io.github.freya022.botcommands.api.commands.text.TextCommandFilter
import io.github.freya022.botcommands.api.commands.text.TextCommandVariation
import io.github.freya022.botcommands.api.core.service.annotations.BService
import io.github.freya022.botcommands.test.switches.TestLanguage
import io.github.freya022.botcommands.test.switches.TestService
import net.dv8tion.jda.api.events.message.MessageReceivedEvent

@BService
@TestService
@TestLanguage(TestLanguage.Language.KOTLIN)
class MyTextCommandFilter : TextCommandFilter {

    override val global: Boolean get() = true

    override suspend fun checkSuspend(
        event: MessageReceivedEvent,
        commandVariation: TextCommandVariation,
        args: String
    ): String? {
        if (event.channel.idLong != 722891685755093076) {
            event.message.reply("Can only run commands in <#722891685755093076>").await()
            return "Wrong channel"
        }
        return null
    }
}
